third_party/terraform/data_sources/data_source_sql_backup_run.go

```diff
@@ -0,0 +1,106 @@
+package google
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	sqladmin "google.golang.org/api/sqladmin/v1beta4"
+)
+
+func dataSourceSqlBackupRun() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceSqlBackupRunRead,
+
+		Schema: map[string]*schema.Schema{
+			"backup_id": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: `The identifier for this backup run. Unique only for a specific Cloud SQL instance. If left empty and multiple backups exist for the instance, most_recent must be set to true.`,
+			},
+			"instance": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Name of the database instance.`,
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `Location of the backups.`,
+			},
+			"start_time": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The time the backup operation actually started in UTC timezone in RFC 3339 format, for example 2012-11-15T16:19:00.094Z.`,
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The status of this run.`,
+			},
+			"most_recent": {
```

Comment:
For context, the AWS equivalent behavior is shown here: https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/data_source_aws_db_snapshot.go#L161